### PR TITLE
Switch from Blacklisting to Whitelisting

### PR DESCRIPTION
**Pull request by mariano-aguero**
This is a proposal to change from blacklisting to whitelisting. See #73
We can discuss the solution raised in this pull request, ideas are welcome.

Changes made:
- Update rules to have separation of concepts and single responsibility principle in the lib folder
- Add logic to manage whitelisting
- Update test to support whitelisting
- Refactoring of test files to match rule id

Regards

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
   "homepage": "https://protofire.github.io/solhint/",
   "repository": "https://github.com/protofire/solhint",
   "scripts": {
-    "exec-tests": "nyc -x \"**/grammar/**\" mocha && nyc report --reporter=text-lcov | coveralls",
+    "exec-tests": "nyc -x \"**/grammar/**\" mocha --recursive \"./test/rules/**/*.js\" && nyc report --reporter=text-lcov | coveralls",
```
Comment (fvictorio) [inline package.json]: Why is this necessary?

Comment (mariano-aguero) [reply package.json]: Mocha searches for the files in the test directory. Mocha does not find the tests in the subfolder rules. I think with the recursive option it would be fine

**Comment by mariano-aguero** [comment]
There is a mayor breaking change so we changed the base branch to 2.0

Regards

**Comment by mariano-aguero** [comment]
> This looks good, but just to clarify: this PR does not add the rulesets feature, or am I missing something here?

It will be added in a next PR

Regards

**Comment by pablofullana** [comment]
@mariano-aguero looks good. But, given:

> > This looks good, but just to clarify: this PR does not add the rulesets feature, or am I missing something here?

> It will be added in a next PR

Let's just change the PR name to match actual changes.
